### ISSUE TITLE
influxdb 0.9.0-rc3 fails to start

### ISSUE THREAD

**Issue by hawson** [influxdb/influxdb#1713]
A checkout and build of 0.9.0-rc3 fails to start with a "panic: runtime error: invalid memory address or nil pointer dereference" error.

The full details are here: https://gist.github.com/hawson/ae53ad40dac7dcb5cf62

I think I've built it properly, but am not 100% certain about that.  I'm using golang 1.4 on a Centos 6.6 system.

**Comment by otoolep** [comment]
Please show us how you are starting `influxd` including any config file passed to it.

**Comment by otoolep** [comment]
@toddboom -- from looking at the code, the reporting code is unconditionally referencing the `b` object. It should instead only do this is `b` is not nil. This is not guaranteed. Specifically if the node is not a Raft node (broker) be *will* be nil.

**Comment by hawson** [comment]
I also just tried the files from the RPM package (manually extracted and run--we have own own locations for external programs), and got much the same error.

The startup boils down to this ("daemon" being the standard function provided by /etc/init.d/functions on RHEL/CentOS hosts):

    user=influxdb
    pidfile=/var/run/influx.pid
    config=/etc/influxdb/influxdb.conf
    LOG=/var/log/influxdb/influx.log
    daemon --user $user --pidfile $pidfile $daemon run -config $config > $LOG 2>&1 &

The config file is here:  https://gist.github.com/hawson/49c25469542d839236b3

**Comment by hawson** [comment]
Also, this is a single-node install:  no replication or clustering happening.

**Comment by hawson** [comment]
Disabling reporting ("reporting-disabled = true") allows the daemon to start.

**Comment by otoolep** [comment]
At first pass, I cannot reproduce this with a single node. Can you also show me your log file?

**Comment by toddboom** [comment]
@hawson if you have any data from a previous RC, please remove it. it's probably located in `~/.influxdb`.

**Comment by toddboom** [comment]
@otoolep I don't think `b` can be nil in the current release, so i think we're ok there.

**Comment by hawson** [comment]
This is is a new install; there is no old data.  The output log contains:

    # cat influxd.log

     8888888           .d888 888                   8888888b.  888888b.
       888            d88P"  888                   888  "Y88b 888  "88b
       888            888    888                   888    888 888  .88P
       888   88888b.  888888 888 888  888 888  888 888    888 8888888K.
       888   888 "88b 888    888 888  888  Y8bd8P' 888    888 888  "Y88b
       888   888  888 888    888 888  888   X88K   888    888 888    888
       888   888  888 888    888 Y88b 888 .d8""8b. 888  .d88P 888   d88P
     8888888 888  888 888    888  "Y88888 888  888 8888888P"  8888888P"
    
    [raft]  2015/02/24 17:33:02 log pending: waiting for initialization or join
    [srvr] 2015/02/24 17:33:02 broker listening on 0.0.0.0:8086
    [srvr] 2015/02/24 17:33:02 broker enforcing retention policies with check     interval of 10m0s
    [srvr] 2015/02/24 17:33:02 data node #0 listening on 0.0.0.0:8086
    [srvr] 2015/02/24 17:33:02 starting admin server on :8083


I redirect STDERR to the file as well, so the go panic messages then follow (as posted in the original gist).

> @otoolep I don't think b can be nil in the current release, so i think we're ok there.

I think I'm running 0.9.0-rc3 (and/or git master branch) and have this problem; is that not the current release?

**Comment by otoolep** [comment]
Your log is different from the log I see if I run a brand new node. You a) are using old data or b) are not running the release you think you are.

**Comment by otoolep** [comment]
```
$ ./influxd 

 8888888           .d888 888                   8888888b.  888888b.
   888            d88P"  888                   888  "Y88b 888  "88b
   888            888    888                   888    888 888  .88P
   888   88888b.  888888 888 888  888 888  888 888    888 8888888K.
   888   888 "88b 888    888 888  888  Y8bd8P' 888    888 888  "Y88b
   888   888  888 888    888 888  888   X88K   888    888 888    888
   888   888  888 888    888 Y88b 888 .d8""8b. 888  .d88P 888   d88P
 8888888 888  888 888    888  "Y88888 888  888 8888888P"  8888888P"

[srvr] 2015/02/24 09:38:37 No config provided, using default settings
[raft]  2015/02/24 09:38:37 log pending: waiting for initialization or join
[raft] marx:8086 2015/02/24 09:38:37 log state change: stopped => leader
[raft] marx:8086 2015/02/24 09:38:37 log initialize: promoted to 'leader' with cluster ID 5705587654662456858, log ID 1, term 1
[srvr] 2015/02/24 09:38:37 broker listening on :8086
[messaging] 2015/02/24 09:38:37 connected to broker: http://marx:8086/messaging/messages?replicaID=1
[srvr] 2015/02/24 09:38:37 broker enforcing retention policies with check interval of 10m0s
[srvr] 2015/02/24 09:38:37 data node #1 listening on :8086
[srvr] 2015/02/24 09:38:37 starting admin server on :8083
[srvr] 2015/02/24 09:38:37 Sending anonymous usage statistics to m.influxdb.com
```

**Comment by otoolep** [comment]
I'd like to see the output when you invoke the command manually.

**Comment by hawson** [comment]
It would be very useful for the daemon to report it's version in the logs when it starts.  The CLI client does this, which is good.

It's entirely possible I'm not running the version I think I am.  I'll purge everything and start over.

**Comment by otoolep** [comment]
You read my mind @hawson:

https://github.com/influxdb/influxdb/pull/1716

**Comment by hawson** [comment]
> You read my mind @hawson:

Unfortunately, my ESP module is *really* unstable, and has a tendency to cause brain core dumps.  Those are really messy to clean up. :-(

**Comment by hawson** [comment]
After nuking my $GOPATH directory and rebuilding as per https://github.com/influxdb/influxdb/blob/master/CONTRIBUTING.md , I tried again to create a database.  here are the log entries

```
# cat influxd.log

 8888888           .d888 888                   8888888b.  888888b.
   888            d88P"  888                   888  "Y88b 888  "88b
   888            888    888                   888    888 888  .88P
   888   88888b.  888888 888 888  888 888  888 888    888 8888888K.
   888   888 "88b 888    888 888  888  Y8bd8P' 888    888 888  "Y88b
   888   888  888 888    888 888  888   X88K   888    888 888    888
   888   888  888 888    888 Y88b 888 .d8""8b. 888  .d88P 888   d88P
 8888888 888  888 888    888  "Y88888 888  888 8888888P"  8888888P"

[raft]  2015/02/24 17:52:04 log pending: waiting for initialization or join
[srvr] 2015/02/24 17:52:04 broker listening on 0.0.0.0:8086
[srvr] 2015/02/24 17:52:04 broker enforcing retention policies with check interval of 10m0s
[srvr] 2015/02/24 17:52:04 data node #0 listening on 0.0.0.0:8086
[srvr] 2015/02/24 17:52:04 starting admin server on :8083
[http] 127.0.0.1 - - [24/Feb/2015:17:52:38 +0000] GET /ping HTTP/1.1 204 0 - InfluxDBShell/0.9 ec2d74d0-bc4d-11e4-a13c-b8ca3a5e3760
[http] 127.0.0.1 - - [24/Feb/2015:17:52:42 +0000] GET /query?db=&q=create+database+storage HTTP/1.1 500 60 - InfluxDBShell/0.9 ee1f32e9-bc4d-11e4-a13c-b8ca3a5e3760
[http] 127.0.0.1 - - [24/Feb/2015:17:52:48 +0000] GET /query?db=foo&q=create+database+storage HTTP/1.1 500 60 - InfluxDBShell/0.9 f1fc0d16-bc4d-11e4-a13c-b8ca3a5e3760
```

**Comment by hawson** [comment]
I just updated the master branch, and pulled in the 'print version' commit, so I have a line:

    [srvr] 2015/02/24 17:58:22 influxdb started, version 0.9, commit unknown

the log is otherwise the same.

So I'm more confident I'm running the proper code now as well (plus `go build -x` shows the expected `cp` commands to put new binaries in their proper location).

However, when `reporting-disabled = false`, I still get a panic.  When `reporting-disabled = true`, the daemon starts.

**Comment by otoolep** [comment]
You are not blowing away your Raft directory. Please remove the directory `/export/home/influxdb/` and try again.

**Comment by hawson** [comment]
There's nothing in the raft dir, except for another directory:
```
[influxdb@storstat11 influxdb]$ find /export/home/influxdb/shared/ -ls
19267596    4 drwxr-xr-x   3 influxdb influxdbgrp     4096 Feb 20 22:46 /export/home/influxdb/shared/
19267587    4 drwxr-xr-x   5 influxdb influxdbgrp     4096 Feb 20 22:46 /export/home/influxdb/shared/data
19267598    4 drwxr-xr-x   3 influxdb influxdbgrp     4096 Feb 23 17:22 /export/home/influxdb/shared/data/db
19270327   24 -rw-------   1 influxdb influxdbgrp  1048576 Feb 24 14:59 /export/home/influxdb/shared/data/db/meta
19401714    4 drwx------   2 influxdb influxdbgrp     4096 Feb 23 17:22 /export/home/influxdb/shared/data/db/shards
19267599    4 drwxr-xr-x   3 influxdb influxdbgrp     4096 Feb 23 17:22 /export/home/influxdb/shared/data/raft
19270326    4 drwx------   2 influxdb influxdbgrp     4096 Feb 23 17:22 /export/home/influxdb/shared/data/raft/raft
19267600    4 drwxr-xr-x   2 influxdb influxdbgrp     4096 Feb 20 22:46 /export/home/influxdb/shared/data/wal
```

* Removing the one file--`data/db/meta`--and restarting didn't do anything.
* Removing the raft directories also didn't change anything. (nuked `meta` as well)
* Removing _everything_ under `shared` *does* seem to fix things....hmm...

Ah, I bet it's because only part of the directory tree is created automatically (puppet).  I've pulled those rules, so only the top-level `shared` directory is created (note the lack of an id, config or term file above) .

And....I think that this also resolves #1714.

I think this one can get closed out: PEBKAC.
